dcap/quoteverify/src/verify.rs

```diff
@@ -2,8 +2,43 @@
 
 //! This module contains logic to verify a DCAP quote
 
+use mc_sgx_core_types::{BaseQuote, Quote};
+use mc_sgx_dcap_sys_types::{sgx_ql_qv_result_t, time_t};
 use mc_sgx_dcap_types::Quote3Error;
 use mc_sgx_util::ResultInto;
+use std::{
+    ptr,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+pub trait Verify: BaseQuote {
+    fn verify(&self) -> Result<(), Quote3Error> {
```

Comment:
This is _minimal_ exercising of the verification functionality.  I'm up for suggestions on how to further extend.  It feels like a builder pattern might be nice `with_supplemental().with_collateral().with_report_info().verify()`

Reply:
I agree that a builder pattern is the way to go here.